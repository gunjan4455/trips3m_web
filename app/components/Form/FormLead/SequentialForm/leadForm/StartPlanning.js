import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';

import Footer from '../footer';
import {
  SEQUENTIAL_FORM_TWO,
} from 'constants/Slider';
import { requiredWithmessage } from "../../../../../utils/formValidators";
import { Destination } from "../../../../../helpers/Icon/Icon";
import { renderToLocation, genericCheckbox } from '../../FormOneVariation';
import { FUNNEL_UUID_KEY, getFormType, getLeadFunnelName, LEAD_FUNNEL, trackSegment } from 'actions/segmentEvents';
import { getCookie } from 'helpers/FormDataFormatter';

const suggestedDestinations = ['Kerala', 'Thailand', 'Himachal', 'Bali', 'Kashmir', 'Maldives', 'Andaman',
  'Mauritius', 'Sikkim', 'Singapore', 'Australia'];

class StartPlanning extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showDestinations: false,
      showIcon: true
    };
  }

  goBack = () => {
    return false;
  };

  goForward = () => {
    const { formValues } = this.props;
    trackSegment({
      event: LEAD_FUNNEL,
      category: `Destination:${formValues.to_loc || 'unknown'}`,
      section: '',
      cta: 'next',
      form_variant: getFormType(formValues.formType),
      funnel_step: 2,
      to_loc: formValues.to_loc,
      label: getLeadFunnelName(2),
      funnel_uuid: getCookie(FUNNEL_UUID_KEY)
    });
    this.props.callNextSlide({
      submitVariation: '',
      variation: true,
      variationStep: SEQUENTIAL_FORM_TWO,
      selectedPillIndex: 0
    });
  };

  handleChangeInput = (destination, i) => {
    this.setState({ selectedPillIndex: i });
    this.props.changeFormField('LeadForm', 'to_loc', destination);
    this.setState({ showDestinations: false, showIcon: true });
  };

  renderSuggestedPills = () => {
    return (
      <div className="row row- mb15">
        <div className="destination-suggestions flex overflowa">
          {suggestedDestinations.map((destination, i) => (
            <div className={`suggested-item sfc1 f14 mr8 ${this.state.selectedPillIndex === i ? 'active' : ''}`}
                 key={i}
                 onClick={() => this.handleChangeInput(destination, i)}>{destination}</div>
          ))}
        </div>
      </div>
    );
  };

  onToLocFocus = () => {
    this.setState({ showDestinations: true, showIcon: false });
  };

  handleBlur = () => {
    this.setState({ showDestinations: false, showIcon: false });
    this.props.handleBlur();
  };

  isAndroid = () => {
    if ( navigator && navigator.userAgent ) {
      return navigator.userAgent.toLowerCase().indexOf("android") > -1 ;
    }
    return false;
  };

  onFromLocFocus = (event) => {
    const elem = event.target;
    if ( this.isAndroid() && elem ) {
      setTimeout(()=> {
        window.scrollTo(0, elem.getBoundingClientRect().top+140);
      },500);

    }
  };

  render() {
    const goNextDisabled = !this.props.formValues.to_loc;
    return (
      <div className="form-body">
        <div className="location-icon-container">
          <Destination />
        </div>
        <div className="form-field-incl-label p15 relative">
          <Field
            name="to_loc"
            component={renderToLocation}
            placeholder="To"
            type="text"
            label="To"
            validate={requiredWithmessage('Please enter your destination')}
            trackFormOneInteracted={this.props.handlFormInteraction}
            onToLocFocus={this.onToLocFocus}
            handleBlur={this.handleBlur}
          />
          <div className="row row- mb15">
            <Field
              name="exploring_destinations"
              component={genericCheckbox}
              type="checkbox"
              label="I am exploring destinations"
              className="checkbox-common sfcw"
              id="exploring-destination_fnew"
              trackFormOneInteracted={this.props.handlFormInteraction}
              classname="exploring-destination at_explore mb8"
            />
          </div>
          {/*{this.state.showDestinations && this.renderSuggestedPills()}*/}
          <div className="from-location-container">
          <Field
            id="fromLocation_fnew"
            className="input"
            name="from_loc"
            component={this.props.FromLocation}
            placeholder="From"
            type="text"
            label="From"
            trackFormOneInteracted={this.props.handlFormInteraction}
            onFocus={this.onFromLocFocus}
          />
          </div>
        </div>
        <Footer goForward={this.goForward} goBackDisabled={true} goNextDisabled={goNextDisabled} />
      </div>
    );
  }
}

StartPlanning.propTypes = {
  formStage: PropTypes.number,
  FromLocation: PropTypes.func,
  changeFormField: PropTypes.func,
  handlFormInteraction: PropTypes.func,
  handleBlur: PropTypes.func,
  formValues: PropTypes.object,
  requiredWithmessage: PropTypes.func,
  callNextSlide: PropTypes.func,
  progressBar: PropTypes.func,
  sliderDisplayMode: PropTypes.string
};

export default StartPlanning;
