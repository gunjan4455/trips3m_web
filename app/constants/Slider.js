const SLIDER_1 = 1;
const SLIDER_2 = 2;
const SLIDER_3 = 3;
const DEALS = 'DEALS';
const VARIATION_1 = 'VARIATION_1';
const VARIATION_2 = 'VARIATION_2';
const VARIATION_3 = 'VARIATION_3';
const FORM_ONE_PART_ONE = 'FORM_ONE_PART_ONE';
const FORM_ONE_PART_TWO = 'FORM_ONE_PART_TWO';
const SEQUENTIAL_FORM_ONE = 'SEQUENTIAL_FORM_ONE';
const SEQUENTIAL_FORM_TWO = 'SEQUENTIAL_FORM_TWO';
const SEQUENTIAL_FORM_THREE = 'SEQUENTIAL_FORM_THREE';
const SEQUENTIAL_FORM_FOUR = 'SEQUENTIAL_FORM_FOUR';
const SEQUENTIAL_FORM_FIVE = 'SEQUENTIAL_FORM_FIVE';
const SEQUENTIAL_FORM_SIX = 'SEQUENTIAL_FORM_SIX';
const SEQUENTIAL_FORM_SEVEN = 'SEQUENTIAL_FORM_SEVEN';
const SEQUENTIAL_FORM_END = 'SEQUENTIAL_FORM_END';
const adminCommentsHashTag = '#slider_organic';
const formFilledOn = '#Form_filled_on';
const publicProductHashTag = '#pp_desktop';
const UPDATE_REDIS_KEY = 'updateFields/UPDATE_REDIS_KEY';
const UPDATE_TO_DESTINATION = 'updateFields/UPDATE_TO_DESTINATION';
const SET_DISPLAY_MODE = 'updateFields/SET_DISPLAY_MODE';
const UPDATE_FIELD = 'updateFields/UPDATE_FIELD';
const NEXT_SLIDE = 'nextSlide';
const PREV_SLIDE = 'prevSlide';
const ENABLED_FOR_SUBMIT = 'slider/ENABLED_FOR_SUBMIT';
const SLIDER_2_SUBMIT = 'slider_2_submit';
const TITLE = {
  1: 'Your Preferences',
  2: 'Great! Tell Us What You Prefer',
  3: 'Almost Done!'
};
const TRIP_UPDATED_IN_REDIS = "updateFields/TRIP_UPDATED_IN_REDIS";

export {
  SLIDER_1,
  SLIDER_2,
  SLIDER_3,
  adminCommentsHashTag,
  formFilledOn,
  publicProductHashTag,
  UPDATE_REDIS_KEY,
  UPDATE_TO_DESTINATION,
  SET_DISPLAY_MODE,
  UPDATE_FIELD,
  NEXT_SLIDE,
  PREV_SLIDE,
  ENABLED_FOR_SUBMIT,
  SLIDER_2_SUBMIT,
  TITLE,
  VARIATION_1,
  VARIATION_2,
  VARIATION_3,
  FORM_ONE_PART_ONE,
  FORM_ONE_PART_TWO,
  SEQUENTIAL_FORM_ONE,
  SEQUENTIAL_FORM_TWO,
  SEQUENTIAL_FORM_THREE,
  SEQUENTIAL_FORM_FOUR,
  SEQUENTIAL_FORM_FIVE,
  SEQUENTIAL_FORM_SIX,
  SEQUENTIAL_FORM_SEVEN,
  SEQUENTIAL_FORM_END,
  TRIP_UPDATED_IN_REDIS,
  DEALS
};
