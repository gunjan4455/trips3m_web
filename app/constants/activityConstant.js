export const ACTIVITY = Object.freeze({
  flight: 'flight',
  itinerary: 'itinerary',
  cab: 'cab',
  hotel: 'hotel',
  attraction: 'attraction'
});
