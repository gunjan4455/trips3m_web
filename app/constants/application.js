export const TT_EMAIL = 'customercare@traveltriangle.com';
export const TT_PHONE = '18001235555';
