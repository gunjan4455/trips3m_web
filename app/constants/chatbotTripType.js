export default [
  {
    id : 'TripType',
    optionId: 1,
    value: 'I am looking for a romantic trip.',
    label: 'Yes! A romantic trip',
    message: 'Yes! A romantic trip',
    trigger: 'Assistance Type Module',
    rbtKey: 'tour',
    rbtValue: 'romantic',
    apiFlag: false,
    dynamicFlag: false,
    user: true
  },
  {
    id : 'TripType',
    optionId: 2,
    value: 'I am looking for a family trip.',
    label: 'Yes! For a family trip',
    message: 'Yes! For a family trip',
    trigger: 'Assistance Type Module',
    rbtKey: 'tour',
    rbtValue: 'family',
    apiFlag: false,
    dynamicFlag: false,
    user: true
  },
  {
    id : 'TripType',
    optionId: 3,
    value: 'I am looking for a honeymoon trip.',
    label: 'Yes! A honeymoon trip',
    message: 'Yes! A honeymoon trip',
    trigger: 'Assistance Type Module',
    rbtKey: 'tour',
    rbtValue: 'honeymoon',
    apiFlag: false,
    dynamicFlag: false,
    user: true
  },
  {
    id : 'TripType',
    optionId: 4,
    value: 'I am looking for a trip with my friends.',
    label: 'Yes! For a trip with my friends',
    message: 'Yes! For a trip with my friends',
    trigger: 'Assistance Type Module',
    rbtKey: 'tour',
    rbtValue: 'friends',
    apiFlag: false,
    dynamicFlag: true,
    user: true
  },
  {
    id : 'TripType',
    optionId: 5,
    value: 'I am looking for a group trip.',
    label: 'For a group trip',
    message: 'For a group trip',
    trigger: 'Assistance Type Module',
    rbtKey: 'tour',
    rbtValue: 'group',
    apiFlag: false,
    dynamicFlag: true,
    user: true
  },
  {
    id : 'TripType',
    optionId: 6,
    value: 'I am looking for a solo trip.',
    label: 'For a solo trip',
    message: 'For a solo trip',
    trigger: 'Assistance Type Module',
    rbtKey: 'tour',
    rbtValue: 'solo',
    apiFlag: false,
    dynamicFlag: true,
    user: true
  },
];
