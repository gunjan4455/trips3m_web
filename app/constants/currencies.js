export const CurrencyList = Object.freeze([
  { label: 'INR', value: 'INR'},
  { label: 'USD', value: 'USD'},
  { label: 'EUR', value: 'EUR'},
  { label: 'SGD', value: 'SGD'},
  { label: 'MYR', value: 'MYR'},
  { label: 'PHP', value: 'PHP'},
  { label: 'GBP', value: 'GBP'},
  { label: 'AUD', value: 'AUD'},
  { label: 'CAD', value: 'CAD'},
  { label: 'SAR', value: 'SAR'},
  { label: 'LBP', value: 'LBP'},
  { label: 'ZAR', value: 'ZAR'},
  { label: 'PKR', value: 'PKR'},
  { label: 'AED', value: 'AED'},
  { label: 'CHF', value: 'CHF'},
  { label: 'JPY', value: 'JPY'},
  { label: 'NZD', value: 'NZD'}
]);

export const DEFAULT_CURRENCY = Object.freeze({ label: 'INR', value: 'INR'});
