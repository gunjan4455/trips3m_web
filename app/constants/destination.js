export const country ={
  INDIA:'india'
};
