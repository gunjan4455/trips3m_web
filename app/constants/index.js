const DEFAULT_IMAGE_PATH = 'https://traveltriangle.com/assets/city_tour_guide_mobile/default_destination-8a3032af47bb4bba85c8148881953d38.png';

export {
  DEFAULT_IMAGE_PATH
};
