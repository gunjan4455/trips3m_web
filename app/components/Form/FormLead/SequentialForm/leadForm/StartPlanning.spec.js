import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import StartPlanning from './StartPlanning';
import { SEQUENTIAL_FORM_TWO } from "../../../../../constants/Slider";

configure({ adapter: new Adapter() });

const mockHandleBlur = jest.fn();
const mockChangeFormField = jest.fn();
const mockCallNextSlide = jest.fn();

const DEFAULT_PROPS = {
  formValues: {
    email : '',
    dont_utilize_budget_new : true,
    budget_new : true,
    start_date_noidea: false,
    adult : '2 ',
    child: 0,
    isd_code: ' + 91 ',
    child_age_answers_attributes: [
      {
        user_input: '0 ',
        customization_id: '7 ',
        option_id: '29 '
      },
      {
        user_input: '0 ',
        customization_id: '7 ',
        option_id: '30 '
      },
      {
        user_input: '0 ',
        customization_id: '7 ',
        option_id: '31 '
      }
    ],
    syncErrors: {},
    to_loc: ''
  },
  handleBlur: mockHandleBlur,
  changeFormField: mockChangeFormField,
  callNextSlide: mockCallNextSlide
};

const shallowWrapperStartPlanning = (props) => {
  return shallow(<StartPlanning {...props} />);
};

describe('lead form module', () => {

  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should render correctly', () => {
    const wrapper = shallowWrapperStartPlanning(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onToLocFocus properly', () => {
    const wrapper = shallowWrapperStartPlanning(props);
    wrapper.instance().onToLocFocus();
    expect(wrapper.state().showDestinations).toEqual(true);
    expect(wrapper.state().showIcon).toEqual(false);
  });

  it('should call handleBlur properly', () => {
    const wrapper = shallowWrapperStartPlanning(props);
    wrapper.instance().handleBlur();
    expect(wrapper.state().showDestinations).toEqual(false);
    expect(wrapper.state().showIcon).toEqual(false);
    expect(mockHandleBlur).toHaveBeenCalled();
  });

  it('should call handleChangeInput properly', () => {
    const wrapper = shallowWrapperStartPlanning(props);
    wrapper.instance().handleChangeInput('kerala', 1);
    expect(wrapper.state().showDestinations).toEqual(false);
    expect(wrapper.state().showIcon).toEqual(true);
    expect(mockChangeFormField).toHaveBeenCalledWith('LeadForm', 'to_loc', 'kerala');
  });

  it('should call goForward properly', () => {
    const expectedInput = {
      submitVariation: '',
      variation: true,
      variationStep: SEQUENTIAL_FORM_TWO,
      selectedPillIndex: 0
    };
    const wrapper = shallowWrapperStartPlanning(props);
    wrapper.instance().goForward();
    expect(mockCallNextSlide).toHaveBeenCalledWith(expectedInput);
  });

  it('should call goBack properly', () => {
    const wrapper = shallowWrapperStartPlanning(props);
    expect(wrapper.instance().goBack()).toEqual(false);
  });
});
