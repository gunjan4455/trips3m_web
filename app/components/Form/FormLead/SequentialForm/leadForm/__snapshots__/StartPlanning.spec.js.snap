// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lead form module should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StartPlanning
    callNextSlide={[MockFunction]}
    changeFormField={[MockFunction]}
    formValues={
      Object {
        "adult": "2 ",
        "budget_new": true,
        "child": 0,
        "child_age_answers_attributes": Array [
          Object {
            "customization_id": "7 ",
            "option_id": "29 ",
            "user_input": "0 ",
          },
          Object {
            "customization_id": "7 ",
            "option_id": "30 ",
            "user_input": "0 ",
          },
          Object {
            "customization_id": "7 ",
            "option_id": "31 ",
            "user_input": "0 ",
          },
        ],
        "dont_utilize_budget_new": true,
        "email": "",
        "isd_code": " + 91 ",
        "start_date_noidea": false,
        "syncErrors": Object {},
        "to_loc": "",
      }
    }
    handleBlur={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="location-icon-container"
        >
          <IconSvg />
        </div>,
        <div
          className="form-field-incl-label p15 relative"
        >
          <Field
            component={[Function]}
            handleBlur={[Function]}
            label="To"
            name="to_loc"
            onToLocFocus={[Function]}
            placeholder="To"
            type="text"
            validate={[Function]}
          />
          <div
            className="row row- mb15"
          >
            <Field
              className="checkbox-common sfcw"
              classname="exploring-destination at_explore mb8"
              component={[Function]}
              id="exploring-destination_fnew"
              label="I am exploring destinations"
              name="exploring_destinations"
              type="checkbox"
            />
          </div>
          <div
            className="from-location-container"
          >
            <Field
              className="input"
              id="fromLocation_fnew"
              label="From"
              name="from_loc"
              onFocus={[Function]}
              placeholder="From"
              type="text"
            />
          </div>
        </div>,
        <Footer
          goBack={[Function]}
          goBackDisabled={true}
          goForward={[Function]}
          goNextDisabled={true}
          isLastSlide={false}
        />,
      ],
      "className": "form-body",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <IconSvg />,
          "className": "location-icon-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Field
              component={[Function]}
              handleBlur={[Function]}
              label="To"
              name="to_loc"
              onToLocFocus={[Function]}
              placeholder="To"
              type="text"
              validate={[Function]}
            />,
            <div
              className="row row- mb15"
            >
              <Field
                className="checkbox-common sfcw"
                classname="exploring-destination at_explore mb8"
                component={[Function]}
                id="exploring-destination_fnew"
                label="I am exploring destinations"
                name="exploring_destinations"
                type="checkbox"
              />
            </div>,
            <div
              className="from-location-container"
            >
              <Field
                className="input"
                id="fromLocation_fnew"
                label="From"
                name="from_loc"
                onFocus={[Function]}
                placeholder="From"
                type="text"
              />
            </div>,
          ],
          "className": "form-field-incl-label p15 relative",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "handleBlur": [Function],
              "label": "To",
              "name": "to_loc",
              "onToLocFocus": [Function],
              "placeholder": "To",
              "trackFormOneInteracted": undefined,
              "type": "text",
              "validate": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Field
                className="checkbox-common sfcw"
                classname="exploring-destination at_explore mb8"
                component={[Function]}
                id="exploring-destination_fnew"
                label="I am exploring destinations"
                name="exploring_destinations"
                type="checkbox"
              />,
              "className": "row row- mb15",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "checkbox-common sfcw",
                "classname": "exploring-destination at_explore mb8",
                "component": [Function],
                "id": "exploring-destination_fnew",
                "label": "I am exploring destinations",
                "name": "exploring_destinations",
                "trackFormOneInteracted": undefined,
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Field
                className="input"
                id="fromLocation_fnew"
                label="From"
                name="from_loc"
                onFocus={[Function]}
                placeholder="From"
                type="text"
              />,
              "className": "from-location-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "input",
                "component": undefined,
                "id": "fromLocation_fnew",
                "label": "From",
                "name": "from_loc",
                "onFocus": [Function],
                "placeholder": "From",
                "trackFormOneInteracted": undefined,
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "goBack": [Function],
          "goBackDisabled": true,
          "goForward": [Function],
          "goNextDisabled": true,
          "isLastSlide": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="location-icon-container"
          >
            <IconSvg />
          </div>,
          <div
            className="form-field-incl-label p15 relative"
          >
            <Field
              component={[Function]}
              handleBlur={[Function]}
              label="To"
              name="to_loc"
              onToLocFocus={[Function]}
              placeholder="To"
              type="text"
              validate={[Function]}
            />
            <div
              className="row row- mb15"
            >
              <Field
                className="checkbox-common sfcw"
                classname="exploring-destination at_explore mb8"
                component={[Function]}
                id="exploring-destination_fnew"
                label="I am exploring destinations"
                name="exploring_destinations"
                type="checkbox"
              />
            </div>
            <div
              className="from-location-container"
            >
              <Field
                className="input"
                id="fromLocation_fnew"
                label="From"
                name="from_loc"
                onFocus={[Function]}
                placeholder="From"
                type="text"
              />
            </div>
          </div>,
          <Footer
            goBack={[Function]}
            goBackDisabled={true}
            goForward={[Function]}
            goNextDisabled={true}
            isLastSlide={false}
          />,
        ],
        "className": "form-body",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <IconSvg />,
            "className": "location-icon-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Field
                component={[Function]}
                handleBlur={[Function]}
                label="To"
                name="to_loc"
                onToLocFocus={[Function]}
                placeholder="To"
                type="text"
                validate={[Function]}
              />,
              <div
                className="row row- mb15"
              >
                <Field
                  className="checkbox-common sfcw"
                  classname="exploring-destination at_explore mb8"
                  component={[Function]}
                  id="exploring-destination_fnew"
                  label="I am exploring destinations"
                  name="exploring_destinations"
                  type="checkbox"
                />
              </div>,
              <div
                className="from-location-container"
              >
                <Field
                  className="input"
                  id="fromLocation_fnew"
                  label="From"
                  name="from_loc"
                  onFocus={[Function]}
                  placeholder="From"
                  type="text"
                />
              </div>,
            ],
            "className": "form-field-incl-label p15 relative",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "handleBlur": [Function],
                "label": "To",
                "name": "to_loc",
                "onToLocFocus": [Function],
                "placeholder": "To",
                "trackFormOneInteracted": undefined,
                "type": "text",
                "validate": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Field
                  className="checkbox-common sfcw"
                  classname="exploring-destination at_explore mb8"
                  component={[Function]}
                  id="exploring-destination_fnew"
                  label="I am exploring destinations"
                  name="exploring_destinations"
                  type="checkbox"
                />,
                "className": "row row- mb15",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "checkbox-common sfcw",
                  "classname": "exploring-destination at_explore mb8",
                  "component": [Function],
                  "id": "exploring-destination_fnew",
                  "label": "I am exploring destinations",
                  "name": "exploring_destinations",
                  "trackFormOneInteracted": undefined,
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Field
                  className="input"
                  id="fromLocation_fnew"
                  label="From"
                  name="from_loc"
                  onFocus={[Function]}
                  placeholder="From"
                  type="text"
                />,
                "className": "from-location-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "input",
                  "component": undefined,
                  "id": "fromLocation_fnew",
                  "label": "From",
                  "name": "from_loc",
                  "onFocus": [Function],
                  "placeholder": "From",
                  "trackFormOneInteracted": undefined,
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "goBack": [Function],
            "goBackDisabled": true,
            "goForward": [Function],
            "goNextDisabled": true,
            "isLastSlide": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
